server/middleware/registration/index.js

```diff
@@ -0,0 +1,114 @@
+const bcrypt = require('bcrypt');
+const logger = require('simple-node-logger').createSimpleLogger();
+const {
+    HttpStatusCodes: { StatusCodes },
+    setLogDetails,
+    DataException
+} = require('../../utils/constants');
+const mongoUtil = require('../../utils/mongoUtil');
+const { HttpStatusCodes } = require('../../utils/constants');
+const JWT = require('jsonwebtoken');
+const { uuid } = require('uuidv4');
+
+const registrationService = async (req, res) => {
+    let saltRounds = 10;
+    let data, statusCode, collectionObj, userPayload, userId;
+
+    let {
+        firstName,
+        lastName,
+        prefLoc,
+        title,
+        email,
+        userName,
+        password,
+    } = req.body;
+    userId = email;
+
+    password = bcrypt.hashSync(password, saltRounds);
+
+    userPayload = {
+        'email': email,
+        'userId': email,
+        'userName': userName,
+        'password': password,
+        'title': title,
+        'prefLoc': prefLoc,
+        'firstName': firstName,
+        'lastName': lastName,
+    }
+
+    try {
+        collectionObj = req.body.collectionObj ? req.body.collectionObj : await mongoUtil.fetchCollection
+            (process.env.USERS_COLLECTION);
+        data = await collectionObj.insertOne({ ...userPayload });
+        if (!data) {
+            statusCode = StatusCodes.REQUEST_TIMEOUT;
```

Comment:
I'm seeing StatusCodes a lot being repeated... let's deconstruct it in the beginning:

`const { REQUEST_TIMEOUT, OK, BAD_REQUEST, GATEWAY_TIMEOUT } = StatusCodes`